java/core/src/main/java/com/whylogs/api/logger/resultSets/ResultSet.java

```diff
@@ -0,0 +1,41 @@
+package com.whylogs.api.logger.resultSets;
+
+import com.whylogs.core.DatasetProfile;
+import com.whylogs.core.errors.Error;
+import com.whylogs.core.metrics.Metric;
+import com.whylogs.core.views.DatasetProfileView;
+import java.util.Optional;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+/**
+ * A holder object for profiling results.
+ *
+ * <p>A whylogs.log call can result in more than one profile. This wrapper class simplifies the
+ * navigation among these profiles.
+ *
+ * <p>Note that currently we only hold one profile but we're planning to add other kinds of profiles
+ * such as segmented profiles here.
+ */
+@Data
+@NoArgsConstructor
+public abstract class ResultSet {
+
+  // TODO: implement read and write when I make the reader and writer
+
+  public abstract Optional<DatasetProfileView> view();
+
+  public abstract Optional<DatasetProfile> profile();
+
+  // TODO: Come back for ModelPerformanceMetrics
+
+  // Question: why is the python addMetrics when it only adds the one?
+  public void addMetric(String name, Metric<?> metric) throws Error {
```

Comment:
What is `name` naming? (same question on the Python side). What is the `get()` on line 39 getting?

Reply:
hmm I assumed the name was the of the column, but that doesn't make sense. In the python there are no usages for this function found. This might just be deadcode that isn't needed. 

The get is just because this is in an optional.

Reply:
I added some `add_metric()` methods to other classes, but I don't think I did this one. I'm not sure it makes sense to add a metric to a `ResultSet`. Maybe run this by @jamie256 and drop it (from Python too) if it's dead. Or just resting. Or pining for the fjords...

Reply:
Alright I sycned with @jamie256 this is not currently being used but is meant for non-segmented model performance metrics row count, etc. Let's go ahead and leave this in here for that as it will be coming in the future. 

I love the pining the fjords haha

Reply:
The ResultSet.add_metric adds a named metric to the profile (not associated with a column). The name here is whatever you want to key the metric on as this will be a dataset level metric, but we don't need this yet in the Java port, I would leave out things we don't need to log some input and have it create a profile that can be uploaded to WhyLabs.